src/actions/pkg.cr

```diff
@@ -1,44 +1,150 @@
 module DppmRestApi::Actions::Pkg
   extend self
+  ALL_PKGS = ""
+  ONE_PKG  = "/:id"
+  include RouteHelpers
+
+  def build_config_response(package, builder)
+    builder.field package.package do
+      builder.object do
+        package.each_config_key do |key|
+          builder.field key, package.get_config key
+        end
+      end
+    end
+  end
+
+  def clean_unused_packages(context : HTTP::Server::Context)
+    if result = Actions.prefix.clean_unused_packages(confirmation: false) { }
+      return {data: result} if result.any?
+      bug = InternalServerError.new context, "received empty set from Prefix#clean_unused_packages; please report this strange bug"
+      error = NoPkgsToClean.new context
+      error.cause = bug
+      raise error
+    end
+    raise NoPkgsToClean.new context
+  end
+
   # List built packages
+  #
+  # TODO optional pagination
   relative_get nil do |context|
-    if context.current_user? && Actions.has_access? context, Access::Read
-      # TODO: list all built packages
+    if Actions.has_access? context, Access::Read
+      JSON.build context.response do |json|
+        json.object do
+          json.field "data" do
+            json.array do
+              Actions.prefix.each_pkg do |package|
+                json.object do
+                  json.field "package", package.package
+                  json.field "version", package.version
+                  package.each_config_key do |key|
```

Comment:
I don't really see the point for packages. However for applications it will be useful.

Reply:
so leave this off, and we're ready to merge?